lambda/function.go

```diff
@@ -16,13 +16,31 @@ import (
 // Function struct which wrap the Handler
 type Function struct {
 	handler Handler
+	context context.Context
 }
 
 // NewFunction which creates a Function with a given Handler
 func NewFunction(handler Handler) *Function {
 	return &Function{handler: handler}
 }
 
+// NewFunctionWithContext which creates a Function with a given Handler and sets the base Context.
+func NewFunctionWithContext(ctx context.Context, handler Handler) *Function {
```

Comment:
Can this be private? `NewFunction` doesn't really have a reason to be in the public API, so it'd be nice to not increase surface area further.

Reply:
@bmoffatt removed this function. I decided to go for the `http.Request.WithContext` approach instead, except that `WithContext` is not exported on the `Function`. I think that's a little cleaner :+1: